test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/PrefixContainerTest.cs

```diff
@@ -138,6 +138,54 @@ public void ContainsPrefix_HasEntries_PrefixMatch_WithSquareBrace(string prefix)
         }
 
         [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(4)]
+        [InlineData(5)]
+        public void ContainsPrefix_HasEntries_PartialAndPrefixMatch_WithDot(int partialMatches)
+        {
+            // Arrange
+            var keys = new string[partialMatches + 1];
+            for (var i = 0; i < partialMatches; i++)
+            {
+                keys[i] = $"aa[{i}]";
+            }
+            keys[partialMatches] = "a.b"; // Sorted before all "aa" keys.
+            var container = new PrefixContainer(keys);
+
+            // Act
+            var result = container.ContainsPrefix("a");
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(4)]
+        [InlineData(5)]
+        public void ContainsPrefix_HasEntries_PartialAndPrefixMatch_WithSquareBrace(int partialMatches)
```

Comment:
@pranavkm in case it's not clear from $6469, this test fails miserably w/o my changes. Would be fine if ASCII had given `[` a lower value than uppercase letters or case-insensitive comparisons didn't uppercase strings (and `ToLower()` didn't have known problems).